### ISSUE TITLE
missing directories should not be created

### ISSUE THREAD

**Issue by viacoban** [derkork/intellij-leiningen-plugin#20]
lein project configuration has a lot of paths like: src test java resources dev-resources

if some of these are missing (not used) in current lein setup they are still getting created by leiningen idea plugin

**Comment by 256BitChris** [comment]
These directories are the defaults that leiningen uses if not overridden in the project.clj file.  So even if they are missing from project.clj, it does not mean that leiningen isn't using them.

The maven plugin's behavior is to create source, resources, and test directories.  This is probably due to the difficulty in determining whether these directories have been configured explicitly or are the defaults.  Since we try to make this plugin mimic the maven plugin and leiningen defaults to these directories, I'd say the plugin is behaving as intended and shouldn't be fixed.

**Comment by viacoban** [comment]
Chris,

Thanks for replying, but I still think this behaviour needs to be changed, several arguments:
- lein itself will not create missing directories even if those are configured in project.clj
- you can very easily add a folder to idea configuration even if folder does not exist on disk
- creation of folders violates the principle of least surprise, I don't expect an ide plugin to modify my source when I perform a read operation like creating a project

Thanks and take care

**Comment by viacoban** [comment]
I have an even more practical reason why this should be fixed, I have a project with a dev dependency outside the source tree:

``` clj
 :profiles {:dev {:resource-paths [~(str (System/getenv "HADOOP_HOME") "/conf")
                                    ~(str (System/getenv "HBASE_HOME") "/conf")]}})
```

HADOOP_HOME AND HBASE_HOME point somewhere users do not have write access. so now because I'm trying this project from my laptop where I do not have hadoop and hbase installed I get these failures and cannot even open this project.clj in idea:

```
Caused by: java.lang.RuntimeException: Could not create directory: /conf
    at de.janthomae.leiningenplugin.module.ModuleCreationUtils$1.run(ModuleCreationUtils.java:96)
    at com.intellij.openapi.application.RunResult.run(RunResult.java:37)
    at com.intellij.openapi.application.WriteAction$1.run(WriteAction.java:35)
    at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:199)
    at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:702)
    at java.awt.EventQueue.access$400(EventQueue.java:82)
    at java.awt.EventQueue$2.run(EventQueue.java:663)
    at java.awt.EventQueue$2.run(EventQueue.java:661)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.AccessControlContext$1.doIntersectionPrivilege(AccessControlContext.java:87)
    at java.awt.EventQueue.dispatchEvent(EventQueue.java:672)
    at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:673)
    at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:498)
    at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:333)
    at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:296)
    at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:211)
    at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:201)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:196)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:188)
    at java.awt.EventDispatchThread.run(EventDispatchThread.java:122)
Caused by: java.io.IOException: Failed to create directory: /Applications/Leda-IC-123.100.app/bin/conf
    at com.intellij.openapi.vfs.impl.local.LocalFileSystemBase.createChildDirectory(LocalFileSystemBase.java:393)
    at com.intellij.openapi.vfs.newvfs.persistent.PersistentFSImpl.createChildDirectory(PersistentFSImpl.java:436)
    at com.intellij.openapi.vfs.newvfs.impl.VirtualFileSystemEntry.createChildDirectory(VirtualFileSystemEntry.java:382)
    at com.intellij.openapi.vfs.VfsUtil.doCreateDirectoriesIfMissing(VfsUtil.java:586)
    at com.intellij.openapi.vfs.VfsUtil.createDirectoryIfMissing(VfsUtil.java:575)
    at de.janthomae.leiningenplugin.module.ModuleCreationUtils$1.run(ModuleCreationUtils.java:93)
    ... 19 more
```

**Comment by 256BitChris** [comment]
Okay, you've convinced me :-)  I'll eliminate the creation of directories if they do not exist.  Thus, the user will have to create the directories he wants as he needs them.

**Comment by viacoban** [comment]
Thanks Chris, I could send you a patch or you want to do this by yourself?

**Comment by 256BitChris** [comment]
I think I can get it fixed in the next hour or so - should be just a minor tweak to the following method:

```
de.janthomae.leiningenplugin.module.ModuleCreationUtils#addSourceFoldersToContentEntry
```

**Comment by viacoban** [comment]
Thanks Chris, do you know when this and #19 are going to be released?

**Comment by 256BitChris** [comment]
Usually whenever Jan gets a chance to merge it in, build it, and upload it to JetBrains.  If you need it faster, I can put the zip file up somewhere and you can upgrade it manually or you can build it following the instructions on the plugin github page.

**Comment by 256BitChris** [comment]
I've updated Pull Request #21 to contain this fix as well.  Jan will update this when he gets the chance via JetBrains.  If this is holding you up, I could email you the new zip file (it's 6 MB) but you'd need to monitor for when Jan releases and get back to the official released version of the plugin so we can support it against a known repeatable environment.

**Comment by viacoban** [comment]
Thanks a lot Chris, I managed to build and install

**Comment by derkork** [comment]
Uploaded new versions 0.3.1 / 0.2.5 with fixes for #19 and #20 to the plugin repo.
